tests/test_null_byte_insertion.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import synapse.rest.admin
+from synapse.rest.client import account, login, register, room
+
+from tests.unittest import HomeserverTestCase
+
+
+class NullByteInsertionTest(HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        login.register_servlets,
+        account.register_servlets,
+        register.register_servlets,
+        room.register_servlets,
+    ]
+
+    def setUp(self):
+        super().setUp()
+
+    # Note that this test must be run with postgres or else is meaningless,
+    # as sqlite will accept insertion of null code points
+    def test_null_byte(self):
+        self.register_user("alice", "password")
+        access_token = self.login("alice", "password")
+        room_id = self.helper.create_room_as("alice", True, "1", access_token)
+        body = '{"body":"\u0000", "msgtype":"m.text"}'
```

Comment:
I think this is meant to be a dictionary, not a string?

Reply:
Setting it as a dict results in the error `Expected: 200, got: 400, resp: b'{"errcode":"M_UNKNOWN","error":"\'body\' not a string type"}'` so I assumed it needed to be a string?

Reply:
I think `send` sends an `m.text` message, so you would just write `"\u0000"` rather than specify any JSON.